qmi/instruments/wl_photonics/wltf_n.py

```diff
@@ -0,0 +1,334 @@
+
+    Attributes:
+        DEFAULT_RESPONSE_TIMEOUT: Timeout value for waiting responses.
+        ZEROING_WAIT: Time to wait for the motor to go to zero.
+        CMD_TERMINATOR: The terminator added for sending commands.
+        RESPONSE_TERMINATOR: The terminator expected for end of one response.
+    """
+    # Default response timeout in seconds.
+    DEFAULT_RESPONSE_TIMEOUT = 5.0
+    ZEROING_WAIT = 3.0  # In seconds
+    
+    # Command and response terminator characters
+    CMD_TERMINATOR = "\r\n"
+    RESPONSE_TERMINATOR = b"OK\r\n"
+
+    def __init__(self, context: QMI_Context, name: str, transport: str) -> None:
+        """Initialize the instrument driver.
+
+        Parameters:
+            context:    The QMI context for running the driver in.
+            name:       Name for this instrument instance.
+            transport:  QMI transport descriptor to connect to the instrument.
+        """
+        super().__init__(context, name)
+        self._transport = create_transport(transport)
+
+        # Instrument ranges for values. Will be updated in `open()`
+        self._wavelength_range = _WavelengthRange
+        self._steps_range = _StepsRange
+
+        # Current motor position step value. Initialize with 0, it will be updated in `open()`
+        self._motor_position = 0
+
+        # For convenience
+        self.encoding = "ascii"
+        self.response_terminator = self.RESPONSE_TERMINATOR.decode(self.encoding)
+
+    def _write(self, cmd: str) -> None:
+        """Write a command to the instrument and check if any errors or alerts were raised.
+
+        Parameters:
+            cmd: The input command to send.
+        """
+        self._check_is_open()
+        cmd = cmd + self.CMD_TERMINATOR
+        self._transport.write(cmd.encode(self.encoding))
+
+    def _read(self) -> str:
+        """Read data from the instrument buffer."""
+        self._check_is_open()
+        response = self._transport.read_until(self.RESPONSE_TERMINATOR, timeout=self.DEFAULT_RESPONSE_TIMEOUT)
+        return response.decode(self.encoding)
+
+    def _clear_ok(self, cmd: str) -> None:
+        """Clear response 'OK' from the instrument buffer. If the returned value was something else than "OK",
+        the clear method will raise an exception.
+
+        Parameters:
+            cmd: The command that was sent in.
+
+        Raises:
+            QMI_InstrumentException: If the instrument returned another response than "OK".
+
+        """
+        response = self._read()
+        # Check that we got 'OK' as second item in response. The first is the echo of the command sent.
+        if response.split(self.response_terminator)[1] == "":
+            return
+
+        # Else try to check what went wrong
+        raise QMI_InstrumentException(f"Command {cmd} returned {response} instead of OK")
+
+    def _ask_int(self, cmd: str) -> int:
+        """Send a query and return an integer number response."""
```

Comment:
docstring missing returns and parameters

Reply:
Added (also "raises")